crates/biome_json_formatter/src/json/value/array_value.rs

```diff
@@ -13,13 +12,25 @@ impl FormatNodeRule<JsonArrayValue> for FormatJsonArrayValue {
             r_brack_token,
         } = node.as_fields();
 
+        let should_expand = f.comments().has_dangling_comments(node.syntax());
+
         write!(
             f,
             [
                 l_brack_token.format(),
-                group(&soft_block_indent(&elements.format())),
+                group(&soft_block_indent(&format_args![
+                    elements.format(),
+                    format_dangling_comments(node.syntax())
+                ]))
+                .should_expand(should_expand),
```

Comment:
This `should_expand` wasn't here before, but since the only way Dangling comments get added is explicitly through the handler that checks for empty blocks, I believe it's safe to add and won't change any other existing formatting.